sched/wdog/wd_cancel.c

```diff
@@ -105,27 +105,24 @@ int wd_cancel_irq(FAR struct wdog_s *wdog)
 
   /* Make sure that the watchdog is still active. */
 
-  if (WDOG_ISACTIVE(wdog))
-    {
-      bool head = list_is_head(&g_wdactivelist, &wdog->node);
+  bool head = list_is_head(&g_wdactivelist, &wdog->node);
```

Comment:
move before line 94

Reply:
this problem doesn't fix yet, @Fix-Point please create a new patch fix it.